File: src/main/java/org/lanternpowered/server/network/vanilla/message/codec/play/CodecPlayOutSetDifficulty.java
package org.lanternpowered.server.network.vanilla.message.codec.play;

import io.netty.buffer.ByteBuf;
import io.netty.handler.codec.CodecException;

import org.lanternpowered.server.network.message.caching.Caching;
import org.lanternpowered.server.network.message.codec.Codec;
import org.lanternpowered.server.network.message.codec.CodecContext;
import org.lanternpowered.server.network.vanilla.message.type.play.MessagePlayOutSetDifficulty;

@Caching
public final class CodecPlayOutSetDifficulty implements Codec<MessagePlayOutSetDifficulty> {

    @Override
    public ByteBuf encode(CodecContext context, MessagePlayOutSetDifficulty message) throws CodecException {
        return context.byteBufAlloc().buffer(1).writeByte(message.getDifficulty().getInternalId());
    }

    @Override
    public MessagePlayOutSetDifficulty decode(CodecContext context, ByteBuf buf) throws CodecException {
        throw new UnsupportedOperationException();
    }
}

File: src/main/kotlin/org/lanternpowered/server/network/vanilla/message/codec/play/CodecPlayOutSetDifficulty.kt
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.network.vanilla.message.codec.play

import org.lanternpowered.server.network.buffer.ByteBuffer
import org.lanternpowered.server.network.message.codec.Codec
import org.lanternpowered.server.network.message.codec.CodecContext
import org.lanternpowered.server.network.vanilla.message.type.play.MessagePlayOutSetDifficulty
import org.lanternpowered.server.world.difficulty.LanternDifficulty

class CodecPlayOutSetDifficulty : Codec<MessagePlayOutSetDifficulty> {

    override fun encode(context: CodecContext, message: MessagePlayOutSetDifficulty): ByteBuffer {
        return context.byteBufAlloc().buffer().apply {
            writeByte((message.difficulty as LanternDifficulty).internalId.toByte())
            writeBoolean(message.locked)
        }
    }
}
